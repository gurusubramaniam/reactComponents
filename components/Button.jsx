/**
 * Name: Button.jsx
 * jsFiddle: https://jsfiddle.net/x1prmh4c/
 * @param {*} props
 * Description:
 *      Button component to render button tag.
 * Example:
 * Takes the props as parameter in the below mentioned format.
 * {
 *  content: {
 *      buttonLabel : "sampleContent"
 *  },
 *  fieldProps {
 *      headSize : 4 (this will render h4 tag)
 *      buttonLabel : "buttonLabel" (this value buttonLabel is the key which should be present in above content section),
 *      btnDisabled : disabled, (this will disable the button)
 *      customBtnClassName: "buttonclassname" (buttonclassname will be applied to the button's class name in case you want to override the style)
 *  }
 * }
 */
const Button = (props) => {
    const {content = {DefaultButtonText: 'Default Button Text'}, onClick, fieldProps = {}} = props;
    const {name = 'defaultButton', buttonLabel = 'DefaultButtonText', value, customBtnClassName = '', dataAutomationId = '', btnDisabled} = fieldProps;

    return (
        <div className={customBtnClassName}>
            <button
                name={name}
                id={name}
                value={value}
                className= {customBtnClassName}
                data-automation-id={dataAutomationId}
                disabled={btnDisabled}
                onClick={onClick}
            >
                {content[buttonLabel]}
            </button>
        </div>
    );
};

export default Button;
