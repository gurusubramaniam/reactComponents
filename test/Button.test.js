import React from 'react';
import { shallow } from 'enzyme';
import Button from '../components/Button.jsx';
import TestRenderer from 'react-test-renderer';

test ("Test Button Component for the expected output", ()=> {
    const wrapper = TestRenderer.create(<Button />);
    console.log(wrapper.toJSON());
})