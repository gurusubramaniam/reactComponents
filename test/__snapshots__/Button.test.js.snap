// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Button Component for the expected output 1`] = `
<div
  className="vx_btn-group _btn_group"
>
  <button
    className="vx_btn "
    data-automation-id=""
    id="defaultButton"
    name="defaultButton"
  >
    Default Button Text
  </button>
</div>
`;
